packages/playout-gateway/src/tsrHandler.ts

```diff
@@ -740,31 +725,24 @@ export class TSRHandler {
 	 * // @todo: proper atem media management
 	 * /Balte - 22-08
 	 */
-	private uploadFilesToAtem(files: { position: number; path: string }[]) {
-		files.forEach((file) => {
-			this.logger.info('try to load ' + JSON.stringify(file) + ' to atem')
-			this.tsr.getDevices().forEach(async (device) => {
-				if (device.deviceType === DeviceType.ATEM) {
-					const options = device.deviceOptions.options as { host: string }
-					this.logger.info('options ' + JSON.stringify(options))
-					if (options && options.host) {
-						this.logger.info('uploading ' + file.path + ' to ' + options.host + ' in MP' + file.position)
-						const process = cp.spawn(`node`, [
-							`./dist/atemUploader.js`,
-							options.host,
-							file.path,
-							file.position.toString(),
-						])
-						process.stdout.on('data', (data) => this.logger.info(data.toString()))
-						process.stderr.on('data', (data) => this.logger.info(data.toString()))
-						process.on('close', () => {
-							process.removeAllListeners()
-						})
-					} else {
-						throw Error('ATEM host option not set')
-					}
+	private uploadFilesToAtem(files: AtemMediaPoolAsset[]) {
+		this.logger.info('try to load ' + JSON.stringify(files.map((f) => f.path).join(', ')) + ' to atem')
+		this.tsr.getDevices().forEach(async (device) => {
```

Comment:
This was pre-existing but is excessive and a bit broken. `uploadFilesToAtem` is called only when a device has a status change, and we then upload to every atem the assets from the one with the status change..

`uploadFilesToAtem` should be updated to only operate on the device that had the status change (perhaps the only parameter should be `device` or `deviceId`)